SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp

```diff
@@ -152,6 +158,26 @@ SpatialGDK::QueryConstraint UGridBasedLBStrategy::GetWorkerInterestQueryConstrai
 	return Constraint;
 }
 
+FVector2D UGridBasedLBStrategy::GetActorLoadBalancingPosition(const AActor& Actor) const
+{
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(Actor.GetNetDriver());
+	const Worker_EntityId ActorEntityId = NetDriver->PackageMap->GetEntityIdFromObject(&Actor);
+
+	if (ActorEntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		return FVector2D(SpatialGDK::GetActorSpatialPosition(&Actor));
+	}
+
+	const SpatialGDK::ComponentData* PositionComponentPtr =
+		NetDriver->Connection->GetCoordinator().GetView()[ActorEntityId].Components.FindByPredicate(
```

Comment:
Grabbing this from the global view doesn't look the prettiest.

Reply:
Why are we doing a fallback to the Spatial position if we have an Actor instance ? Is it in anticipation of the Strategy worker? There is a lot of special casing done in SpatialGDK::GetActorSpatialPosition, and I am afraid that it could break things in the meantime if the current distributed load balancing system bypasses it.

Reply:
This is spot on - I've chatted with @m-samiec and @samiwh, and they both seem to agree that we'll be probably using Spatial position on the Strategy worker, and I think we've reached a consensus it'd be best if we introduced this behaviour now rather than in the same time as the Strategy worker.

So yeah, it's in anticipation. 

The last point shouldn't be a big concern as Spatial position is written by Unreal workers based on `SpatialGDK::GetActorSpatialPosition` so special casing is still there.

Reply:
Tbh the more problems I encounter, the more I think of excluding this change from this PR.

Reply:
I can see the logic, but I am afraid that various parts of the GDK might not be in agreement with each other. For instance, the load balancing handler iterates over all actors asking what their authoritative worker should be. Characters are supposed to follow their possessing Controller, which get their position from the character they are possessing. So with this change, it opens the possibility of a race, where a controller being at the origin by default before possession will force a character to be load balanced with it, to later load balance itself at the character's authoritative worker, because it took some time (a position update + spatial OS roundtrip) before the PC and character position were synchronized. It will eventually work, but it introduces jittering in the current load balancer.

Reply:
Yeah I agree - this does seem more complex than I'd be comfortable adding in this PR. I'm more in favour of removing this change altogether then, and reintroducing it in another context.